lib/resourcemerge/machineconfig.go

```diff
@@ -100,5 +100,10 @@ func ensureControllerConfigSpec(modified *bool, existing *mcfgv1.ControllerConfi
 		*modified = true
 	}
 
+	if existing.DNS != nil && required.DNS != nil && !equality.Semantic.DeepEqual(existing.DNS, required.DNS) {
```

Comment:
since DNS is *:
if existing.DNS != nil && required.DNS != nil {
    !equality.Semantic.DeepEqual(existing.DNS, required.DNS) {
}

Reply:
unless someone can confirm the order of the &&& eval?

Reply:
Yes. This is typical from the days of good old `C` and `C++`. 
You can see code like this in the Unix OS code:

```
if (ptr != nil && ptr->data == blah) {
}
```

Reply:
Yeah, and the short-circuit logic is relied upon by some of the other checks in this function too.

Reply:
thanks just double checking!